haskell/tools.bzl

```diff
@@ -38,7 +47,7 @@ def _get_build_tool(ctx, tool_name):
   Returns:
     File: Build tool with the name user asked for.
   """
-  for tool in get_build_tools(ctx).to_list():
+  for tool in set.to_list(get_build_tools(ctx)):
```

Comment:
I wonder if we can implement a method that lets us traverse over `set` without explicit to_list? `dict` is traversable I believe (though that may be only with `.items` &c.)